### PR TITLE
Chapter 1 solutions + exercises reorganization

### PR DESCRIPTION
**Pull request by bragov4ik**
This PR contains solutions to some exercises from chapter 1 as well as suggestion for exercises reorganization.
I think it should be easier to navigate when each exercise is represented as a separate file (considering there are tens of exercises in each chapter).

### CODE REVIEW

**Comment by fizruk** [comment]
I do not mind keeping each exercise in a separate file. But right now they are not navigable at all, you need to properly add them to the navigation in `mkdocs.yml` or link to them from the main exercises file. You can check the result by running `mkdocs serve` locally.

**Comment by bragov4ik** [comment]
> I do not mind keeping each exercise in a separate file. But right now they are not navigable at all, you need to properly add them to the navigation in `mkdocs.yml` or link to them from the main exercises file. You can check the result by running `mkdocs serve` locally.

https://github.com/rzk-lang/hottbook/pull/4

File: src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md [hunk 1]
```diff
@@ -6,6 +6,173 @@ This is a literate Rzk file:
...
+
+-- #def id
+--   (A : U)
+--   : A -> A
+--   := \ a -> a
+
+-- #def prod
+--   (A B : U)
+--   : U
+--   := Sigma (_ : A), B
```
Comment (fizruk) [inline src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md]: Remove commented out code. You can refer to these definitions from text, e.g. something like this (I have not checked the validity of links):

```
In this solution, we use [`id`](../04-dependent-function-types.rzk/#define:id) and [`prod`](../05-product-types.rzk.md/#define:prod), defined earlier.
```

File: src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md [hunk 2]
```diff
@@ -6,6 +6,173 @@ This is a literate Rzk file:
...
+```rzk
+#def pr_1_prod
+  (A B : U)
+  : prod A B -> A
+  := \(a, b) -> a
+
+#def pr_2_prod
+  (A B : U)
+  : prod A B -> B
+  := \(a, b) -> b
```
Comment (fizruk) [inline src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md]: These are [already defined](https://rzk-lang.github.io/hottbook/1-foundations/1-type-theory/05-product-types.rzk/#define:pr₁).

File: src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md [hunk 3]
```diff
@@ -6,6 +6,173 @@ This is a literate Rzk file:
...
+
+-- rec_AxB((C: U), (g: A -> B -> C), (p: AxB))
```
Comment (fizruk) [inline src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md]: Use regular (markdown) text (with math support), not code comments.

File: src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md [hunk 4]
```diff
@@ -6,6 +6,173 @@ This is a literate Rzk file:
...
+  : prod A B -> A
+  := \(a, b) -> a
+
+#def pr_2_prod
+  (A B : U)
+  : prod A B -> B
+  := \(a, b) -> b
+
+-- rec_AxB((C: U), (g: A -> B -> C), (p: AxB)) 
+#def rec_prod
```
Comment (fizruk) [inline src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md]: I suggest

```suggestion
#def prod-rec-via-projections
```

or 

```suggestion
#def prod-rec'
```

File: src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md [hunk 5]
```diff
@@ -6,6 +6,173 @@ This is a literate Rzk file:
...
+```
+
+Defining projections through a recursion on $A \times B$
+```rzk
+-- other way
+#def rec_prod_reversed
+  (A B : U)
+  (C : U)
+  : (g : A -> B -> C) -> (p : prod A B) -> C
+  := \ g (a, b) -> g a b
```
Comment (fizruk) [inline src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md]: This is [already defined](https://rzk-lang.github.io/hottbook/1-foundations/1-type-theory/05-product-types.rzk/#define:prod-rec).

File: src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md [hunk 6]
```diff
@@ -6,6 +6,173 @@ This is a literate Rzk file:
...
+
+#def pr_1_prod_reversed
+  (A B : U)
+  : prod A B -> A
+  := rec_prod_reversed A B A (\a b -> a)
+
+#def pr_2_prod_reversed
+  (A B : U)
+  : prod A B -> B
+  := rec_prod_reversed A B B (\a b -> b)
```
Comment (fizruk) [inline src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md]: This is [defined as well](https://rzk-lang.github.io/hottbook/1-foundations/1-type-theory/05-product-types.rzk/#define:pr₁').

File: src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md (hunk 2 above)
Comment (bragov4ik) [reply src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md]: How to write something in subscript? It seems inconvenient to go and copy the definition each time :(

Comment (fizruk) [reply src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md]: I use this VS Code extension: https://marketplace.visualstudio.com/items?itemName=yellpika.latex-input

**Pull request merged**